src/librbd/librbd.cc

```diff
@@ -107,6 +107,39 @@ static auto create_write_raw(librbd::ImageCtx *ictx, const char *buf,
       deleter(new UserBufferDeleter(ictx->cct, aio_completion))));
 }
 
+static int get_iovec_length(const struct iovec *iov, int iovcnt, size_t &len)
+{
+  if (iovcnt <= 0) {
+    return -EINVAL;
+  }
+
+  len = 0;
```

Comment:
I don't think it is strictly necessary to initialize on error but I will move it up.

Reply:
It's always better to initialize and it is actually needed in this case because `len` is used in `tracepoint` call.  I think the only reason you didn't get a compiler warning is that tracing stuff is not compiled in by default.